### PR TITLE
Lifecycle integration tests

### PR DESCRIPTION
**Pull request by spoorthipujariadobe**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: code/android-lifecycle-library/src/main/java/com/adobe/marketing/mobile/lifecycle/LifecycleState.java [hunk 1]
```diff
@@ -222,10 +220,6 @@ Map<String, String> getContextData() {
      */
     @VisibleForTesting
     void updateContextData(final Map<String, String> contextData) {
```
Comment (praveek) [inline code/android-lifecycle-library/src/main/java/com/adobe/marketing/mobile/lifecycle/LifecycleState.java:222]: Mark with @NonNull annotation

File: code/android-lifecycle-library/src/main/java/com/adobe/marketing/mobile/lifecycle/LifecycleState.java [hunk 2]
```diff
@@ -238,10 +232,6 @@ void updateContextData(final Map<String, String> contextData) {
      */
     @VisibleForTesting
     void updatePreviousSessionLifecycleContextData(final Map<String, String> contextData) {
```
Comment (praveek) [inline code/android-lifecycle-library/src/main/java/com/adobe/marketing/mobile/lifecycle/LifecycleState.java:234]: Mark with @NonNull annotation

File: code/integration-tests/src/androidTest/java/com/adobe/marketing/mobile/lifecycle/LifecycleIntegrationTests.kt [hunk 3]
```diff
@@ -0,0 +1,400 @@
...
+    }
+
+    @Test
+    fun testInstall() {
+        // setup
+        val countDownLatch = CountDownLatch(1)
+        networkMonitor = { url ->
+            if (url.startsWith("https://www.lifecycle.com")) {
+                // verify
+                assertTrue(url.contains("installevent=InstallEvent"))
```
Comment (praveek) [inline code/integration-tests/src/androidTest/java/com/adobe/marketing/mobile/lifecycle/LifecycleIntegrationTests.kt:109]: Can you verify if assert failure actually fails the test as this happens in background thread?

File: code/integration-tests/src/androidTest/java/com/adobe/marketing/mobile/lifecycle/LifecycleIntegrationTests.kt [hunk 4]
```diff
@@ -0,0 +1,400 @@
...
+            }
+        }
+        MobileCore.updateConfiguration(
+            mapOf(
+                "lifecycle.sessionTimeout" to 300,
+                "global.privacy" to "optedin",
+                "rules.url" to "https://adobe.com/rules_lifecycle.zip"
+            )
+        )
+        Thread.sleep(10)
```
Comment (praveek) [inline code/integration-tests/src/androidTest/java/com/adobe/marketing/mobile/lifecycle/LifecycleIntegrationTests.kt:120]: I would suggest avoiding Thread.sleep and instead use latch.await(...)

File: code/integration-tests/src/androidTest/java/com/adobe/marketing/mobile/lifecycle/LifecycleIntegrationTests.kt (hunk 3 above)
Comment (spoorthipujariadobe) [reply code/integration-tests/src/androidTest/java/com/adobe/marketing/mobile/lifecycle/LifecycleIntegrationTests.kt:109]: The `latch.countdown` on the next line does not run if the assert fails so the test never terminates. Added a timeout to the test.

Comment (praveek) [reply code/integration-tests/src/androidTest/java/com/adobe/marketing/mobile/lifecycle/LifecycleIntegrationTests.kt:109]: Can you capture the variable and assert it outside the block? That way we will easily know about the failure.

**Review by praveek — APPROVED** [review]

**Pull request merged**